src/main/java/domain/Names.java

```diff
@@ -0,0 +1,43 @@
+package domain;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+
+public class Names {
+
+    private static final int MIN_NAME_SIZE = 2;
+
+    private final List<Name> names;
+
+    public Names(List<Name> names) {
+        validate(names);
+        this.names = new ArrayList<>(names);
+    }
+
+    private void validate(List<Name> names) {
+        validateSize(names);
+        validateDuplicate(names);
+    }
+
+    private void validateSize(List<Name> names) {
+        if (names.size() < MIN_NAME_SIZE) {
+            throw new IllegalArgumentException(String.format("이름은 최소 2개 이상이여햡니다. 입력값 : %d", names.size()));
+        }
+    }
+
+    private void validateDuplicate(List<Name> names) {
+        if (new HashSet<>(names).size() != names.size()) {
+            throw new IllegalArgumentException("이름은 중복 될 수 없습니다.");
+        }
+    }
+
+    public int getSize() {
+        return names.size();
+    }
```

Comment:
메서드가 어떤것을 반환할 때, 메서드명은 명사를 사용해도 괜찮습니다 (취향 차이입니다 😄 )

```suggestion
    public int size() {
        return names.size();
    }
```

Reply:
메서드는 반드시 동사를 사용해야한다고 생각했는데 아니었군요.. 제 편견을 깨주셔서 감사합니다. 

특히 일급 컬렉션의 경우엔 명사를 사용할 때가 원래 자바의 컬렉션 함수를 이용하듯 이용할 수 있어 좋겠네요.

훨신 직관적인 코드가 될 수 있을 것 같아요!